spec/controllers/preferences_controller_spec.rb

```diff
@@ -100,12 +84,47 @@ def set_preference(preference, section)
           expect(response.response_code).to eq(201)
         end
 
-        it 'assigns @preference variable' do
-          set_preference(0.145, section)
-          preference = assigns(:preference)
-          expect(preference.preference).to be_within(1e-3).of 0.145
-          expect(preference.user).to eq user
-          expect(preference.section).to eq section
+        describe "if the preference is zero" do
+          describe 'if the record existed' do
+            it 'deletes the record' do
+              expect { set_preference(0, section) }.
+                to change(Preference, :count).by(-1)
+            end
+          end
+
+          describe "if the record didn't exist" do
+            it "doesn't create a new record" do
+              new_section = create(:section_without_course)
+              course.sections << new_section
+              course.save!
+              expect { set_preference(0, new_section) }.
+                to change(Preference, :count).by(0)
+            end
+          end
+        end
+
+        describe "if the preference is not zero" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.